lib/ProductOpener/Images.pm

```diff
@@ -1309,28 +1309,35 @@ sub process_image_crop ($user_id, $product_id, $id, $imgid, $angle, $normalize,
 		$h = $z;
 	}
 
-	print STDERR
-		"image_crop.pl - imgid: $imgid - crop_size: $crop_size - x1: $x1, y1: $y1, x2: $x2, y2: $y2, w: $w, h: $h\n";
-	$log->trace("calculating geometry",
-		{crop_size => $crop_size, x1 => $x1, y1 => $y1, x2 => $x2, y2 => $y2, w => $w, h => $h})
-		if $log->is_trace();
-
-	my $ox1 = int($x1 * $ow / $w);
-	my $oy1 = int($y1 * $oh / $h);
-	my $ox2 = int($x2 * $ow / $w);
-	my $oy2 = int($y2 * $oh / $h);
-
-	my $nw = $ox2 - $ox1;    # new width
-	my $nh = $oy2 - $oy1;
-
-	my $geometry = "${nw}x${nh}\+${ox1}\+${oy1}";
-	$log->debug("geometry calculated",
-		{geometry => $geometry, ox1 => $ox1, oy1 => $oy1, ox2 => $ox2, oy2 => $oy2, w => $w, h => $h})
-		if $log->is_debug();
-	if ($nw > 0) {    # image not cropped
-		my $imagemagick_error = $source->Crop(geometry => $geometry);
-		($imagemagick_error)
-			and $log->error("could not crop to geometry", {geometry => $geometry, error => $imagemagick_error});
+	# avoid a potential divide by zero error
+	if (!$w or !$h) {
+		$log->error("Cannot crop image $imgid , crop width or height is 0: $w x $h");
+	}
+	else {
+
+		print STDERR
+			"image_crop.pl - imgid: $imgid - crop_size: $crop_size - x1: $x1, y1: $y1, x2: $x2, y2: $y2, w: $w, h: $h\n";
+		$log->trace("calculating geometry",
+			{crop_size => $crop_size, x1 => $x1, y1 => $y1, x2 => $x2, y2 => $y2, w => $w, h => $h})
+			if $log->is_trace();
+
+		my $ox1 = int($x1 * $ow / $w);
+		my $oy1 = int($y1 * $oh / $h);
+		my $ox2 = int($x2 * $ow / $w);
+		my $oy2 = int($y2 * $oh / $h);
+
+		my $nw = $ox2 - $ox1;    # new width
+		my $nh = $oy2 - $oy1;
+
+		my $geometry = "${nw}x${nh}\+${ox1}\+${oy1}";
```

Comment:
There's a $geometry on line 1343 below that won't exist if you use "my $geometry" here I think (I may be reading the diff wrong)

Reply:
Oh yes, you're right I misread the `=~` for an `=` so I though it was redefined.